src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs

```diff
@@ -211,39 +210,47 @@ public async Task<List<Guid>> GetApplicationsToExtract(DateTime executionDateTim
 
         public async Task<List<SubmittedApplicationAnswer>> ExtractAnswersForApplication(Guid applicationId)
         {
-            _logger.LogDebug($"Extracting answers for application {applicationId}");
-
+            _logger.LogInformation($"Extracting answers for application {applicationId}");
             var answers = new List<SubmittedApplicationAnswer>();
 
-            var sections = await _qnaApiClient.GetAllSectionsForApplication(applicationId);
-
-            if (sections != null)
+            try
             {
-                foreach (var section in sections)
-                {
-                    var completedPages = section.QnAData.Pages.Where(pg => pg.Active && pg.Complete && !pg.NotRequired);
 
-                    foreach (var page in completedPages)
+                var sections = await _qnaApiClient.GetAllSectionsForApplication(applicationId);
+
+                if (sections != null)
+                {
+                    foreach (var section in sections)
                     {
-                        var submittedPageAnswers = ExtractPageAnswers(applicationId, section.SequenceNo, section.SectionNo, page);
-                        answers.AddRange(submittedPageAnswers);
+                        var completedPages = section.QnAData.Pages.Where(pg => pg.Active && pg.Complete && !pg.NotRequired);
+
+                        foreach (var page in completedPages)
+                        {
+                            var submittedPageAnswers = ExtractPageAnswers(applicationId, section.SequenceNo, section.SectionNo, page);
+                            answers.AddRange(submittedPageAnswers);
+                        }
                     }
                 }
+                //Extract QuestionId Y0-110 answer even if it is inactive, there is an issue setting answer active to false while adding orgnisation type answer to the tabular list as the last entry
+                var hasPartnershipAnswersExtracted = answers.Any(answer => answer.QuestionId == QuestionIdPartnership);
+                if (!hasPartnershipAnswersExtracted)
+                {
+                    await ExtractPartnershipAnswers(applicationId, answers);
+                }
             }
-            //Extract QuestionId Y0-110 answer even if it is inactive, there is an issue setting answer active to false while adding orgnisation type answer to the tabular list as the last entry
-            var hasPartnershipAnswersExtracted = answers.Any(answer => answer.QuestionId == QuestionIdPartnership);
-            if(!hasPartnershipAnswersExtracted)
+            catch (Exception ex)
             {
-                await ExtractPartnershipAnswers(applicationId, answers);
+                _logger.LogError(ex, $"Unable to extract answers for application {applicationId}");
```

Comment:
Remove try catch and use log as discussed before. Or throw the exception.

Reply:
Addressed with throw exception